Framework/API/test/MatrixWorkspaceTest.h

```diff
@@ -1291,6 +1291,97 @@ class MatrixWorkspaceTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(wsCastConst, wsCastNonConst);
   }
 
+  void test_x_uncertainty_can_be_set() {
+    // Arrange
+    WorkspaceTester ws;
+    const size_t numspec = 4;
+    const size_t j = 3;
+    const size_t k = j;
+    ws.init(numspec, j, k);
+
+    const double value0 = 10;
+    const double value1 = 11;
+    const double value2 = 12;
+
+    Mantid::MantidVec dxSpec0(j, value0);
+    Mantid::MantidVecPtr dxSpec1;
+    auto &spec1 = dxSpec1.access();
+    spec1.resize(j, value1);
+    boost::shared_ptr<Mantid::MantidVec> dxSpec2(
+        new Mantid::MantidVec{value2, value2, value2});
+    // Act
+    TSM_ASSERT("Should not have any x resolution values", !ws.hasDx(0));
+    TSM_ASSERT("Should not have any x resolution values", !ws.hasDx(1));
+    TSM_ASSERT("Should not have any x resolution values", !ws.hasDx(2));
+    TSM_ASSERT("Should not have any x resolution values", !ws.hasDx(3));
+
+    ws.setDx(0, dxSpec0);
+    ws.setDx(1, dxSpec1);
+    ws.setDx(2, dxSpec2);
+
+    // Assert
+    // Test dataDx
+    TSM_ASSERT("Should have x resolution values", ws.hasDx(0));
+    TSM_ASSERT_EQUALS("Should have a length of 3", ws.dataDx(0).size(), j);
+    TSM_ASSERT_EQUALS("Should have x resolution values of 10", ws.dataDx(0)[0],
+                      value0);
+    TSM_ASSERT_EQUALS("Should have x resolution values of 10", ws.dataDx(0)[1],
+                      value0);
+    TSM_ASSERT_EQUALS("Should have x resolution values of 10", ws.dataDx(0)[2],
+                      value0);
+
```

Comment:
Should we obtain a reference once, then call the subscript operator multiple times? I just don't want to encourage anyone to repeatedly call `.dataDx()` somewhere where it adds unnecessary overhead.